### PR TITLE
completed react-file-uploads exercise - Evan Hong

### PR DESCRIPTION
**Pull request by EvanS-Hong**

### CODE REVIEW

File: react-file-uploads/client/src/UploadForm.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file
+     */
+    event.preventDefault();
+    const formData = new FormData(event.target);
+
+    const res = fetch('http://localhost:8080/api/uploads', {
+      method: "POST",
+      body: formData,
+  });
+    const jsonData = await res;
```
Comment (RobertGardner) [inline react-file-uploads/client/src/UploadForm.js]: The response comes back in JSON. You should convert it before you log it.

File: react-file-uploads/client/src/UploadForm.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    jsonData === undefined ? console.error('Something went wrong') : console.log(jsonData);
```
Comment (RobertGardner) [inline react-file-uploads/client/src/UploadForm.js:33]: You didn't check the response `ok` field. You don't catch any exceptions that are thrown.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Fix the client error handling.

File: react-file-uploads/client/src/UploadForm.js [hunk 3]
```diff
@@ -21,15 +21,16 @@ export default  function UploadForm() {
...
     const formData = new FormData(event.target);
-
     const res = fetch('http://localhost:8080/api/uploads', {
       method: "POST",
       body: formData,
   });
-    const jsonData = await res;
-
+    if (!res.ok) {
+      throw new Error(`error ${res.status}`);
```
Comment (RobertGardner) [inline react-file-uploads/client/src/UploadForm.js:30]: Almost. This is good now, but you need to use a try/catch to catch exceptions.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: react-file-uploads/client/src/UploadForm.js (hunk 3 above)
Comment (EvanS-Hong) [reply react-file-uploads/client/src/UploadForm.js:30]: oh, I wasn't done, I just pushed because I had to restart VS code and didn't want to lose anything

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**